trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java

```diff
@@ -280,6 +303,66 @@ public void dropCatalog(String metalakeName, String catalogName, boolean ignoreN
     }
   }
 
+  public void alterCatalog(
+      String metalakeName,
+      String catalogName,
+      Map<String, String> setProperties,
+      List<String> removeProperties) {
+    try {
+      NameIdentifier catalogNameId = NameIdentifier.of(metalakeName, catalogName);
+      CatalogConnectorContext catalogConnectorContext =
+          catalogConnectors.get(catalogNameId.toString());
+      GravitinoCatalog oldCatalog = catalogConnectorContext.getCatalog();
+
+      List<CatalogChange> changes = new ArrayList<>();
+      setProperties
+          .entrySet()
+          .forEach(
+              e -> {
+                // Skip the no changed attributes
+                boolean matched =
+                    oldCatalog.getProperties().entrySet().stream()
+                        .anyMatch(
+                            oe ->
+                                oe.getKey().equals(e.getKey())
+                                    && oe.getValue().equals(e.getValue()));
+                if (!matched) {
+                  changes.add(CatalogChange.setProperty(e.getKey(), e.getValue()));
+                }
+              });
+
+      removeProperties.forEach(
+          key -> {
+            if (oldCatalog.getProperties().containsKey(key)) {
+              changes.add(CatalogChange.removeProperty(key));
+            }
+          });
+
+      if (changes.isEmpty()) {
+        return;
+      }
+
+      GravitinoMetalake metalake =
+          gravitinoClient.loadMetalake(NameIdentifier.ofMetalake(metalakeName));
+      metalake.alterCatalog(
+          NameIdentifier.of(metalakeName, catalogName),
+          changes.toArray(changes.toArray(new CatalogChange[0])));
+
+      Future<?> future = executorService.submit(this::loadMetalake);
```

Comment:
I 'm wondering why you 're executing it in another thread here.

Reply:
I hope to perform the catalog operation using a single thread. This way, we don't have to consider thread safety issues.

Reply:
As you will block the main thread and using 
```
future.get(30,....)
``` 

I don't understand why it needs another thread to implement it.

Reply:
Calling `alterCatalog` uses the `query-executer` thread. If we use this thread to perform `alterCatalog`, it will race with the `ScheduledExecutorService` thread that loads the catalog.

Reply:
> it will race with the ScheduledExecutorService thread that loads the catalog.

What will happen if `loadMetalake` and `alterCatalog` at the same time? Don't we support concurrent access between them?

Reply:
Yes, CatalogConnectorManager is not thread-safe. We don't want to make it thread safe, these operations are infrequent.